From 0d04ea01b3f4b283b1709dcf01d9f183d8e3d079 Mon Sep 17 00:00:00 2001
From: "Gu, Wangyi" <wangyi.gu@intel.com>
Date: Tue, 15 Jan 2013 10:01:22 +0800
Subject: [PATCH 07/56] [PATCH] [PORT FROM MAIN] Movie studio: fix a memory
 leak issue in the AOSP code

It's an android defect, connect()/disconnect() of OMXClient need be called in a pair,
if miss calling disconnect() while connect() is called, a memory leak will happen.

Change-Id: I91f6c6af9b6567b4af73fe4cfbed118edcaaccbd
Category: aosp improvement
Domain: Video.Media-video editor
Origin: internal
Signed-off-by: Gu, Wangyi <wangyi.gu@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp b/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
index 5386cdd..93f3cf3 100755
--- a/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
+++ b/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
@@ -779,6 +779,7 @@ M4OSA_ERR queryVideoDecoderCapabilities
     logSupportDecodersAndCapabilities(pDecoders);
     *decoders = pDecoders;
 cleanUp:
+    client.disconnect();
     return err;
 }
 /********************
-- 
2.7.4

